python/src/numerous/collection/numerous_file.py

```diff
@@ -0,0 +1,228 @@
+"""Class for working with numerous files."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, BinaryIO
+
+from numerous.generated.graphql.input_types import TagInput
+
+
+if TYPE_CHECKING:
+    from io import TextIOWrapper
+
+    from numerous.collection._client import Client
+    from numerous.generated.graphql.fragments import CollectionFileReferenceTags
+
+
+_NO_FILE_MSG_ = "File dont exists."
+
+
+class NumerousFile:
+    """
+    Represents a file in a Numerous collection.
+
+    Attributes
+    ----------
+        key (str): The key of the file.
+        collection_info tuple[str, str]: The id
+            and key of collection file belongs to.
+        data (Optional[dict[str, Any]]): The data of the file.
+        id (Optional[str]): The unique identifier of the file.
+        client (Client): The client to connect.
+        numerous_file_tags (dict[str, str]): The tags associated with the file.
+
+    """
+
+    def __init__(
+        self,
+        *,
+        client: Client,
+        key: str,
+        file_id: str,
+        exists: bool = False,
+        numerous_file_tags: list[CollectionFileReferenceTags] | None = None,
+    ) -> None:
+        self.key: str = key
+        self.file_id = file_id
+        self._client: Client = client
+        self._exists = exists
+        self._tags: dict[str, str] = {}
+
+        if numerous_file_tags is not None:
+            dict_of_tags = {tag.key: tag.value for tag in numerous_file_tags}
+            self._tags = dict_of_tags if dict_of_tags else {}
+
+    @property
+    def exists(self) -> bool:
+        """
+        Check if the file exists.
+
+        Returns
+        -------
+            bool: True if the file exists,
+            False otherwise.
+
+        """
+        return self._exists
+
+    @property
+    def tags(self) -> dict[str, str]:
+        """
+        Get the tags associated with the file.
+
+        Returns
+        -------
+            dict[str, str]: Dictionary of tag key-value pairs.
+
+        """
+        return self._tags
+
+    def read_text(self) -> str:
+        """
+        Read the file's content as text.
+
+        Returns
+        -------
+            List[str]: The lines of text from the file.
+
+        Raises
+        ------
+            OSError: If an error occurs while reading the file.
+            ValueError: If there is no local path for the file.
+
+        """
+        if not self.exists:
+            raise ValueError(_NO_FILE_MSG_)
+        return self._client.read_text(self.file_id)
+
+    def read_bytes(self) -> bytes:
+        """
+        Read the file's content as bytes.
+
+        Returns
+        -------
+            bytes: The byte content of the file.
+
+        Raises
+        ------
+            ValueError: If there is no local path for the file.
+
+        """
+        if not self.exists:
+            raise ValueError(_NO_FILE_MSG_)
```

Comment:
The docs for this appears to match the old implementation.

The same is true for other methods. Please update them :)